src/api/session/ui/Imp-base.cpp

```diff
@@ -351,18 +351,18 @@ auto UI::Imp::blockchain_selection(
     const SimpleCallback cb) const noexcept
     -> BlockchainSelectionMap::mapped_type&
 {
-    
+
     auto it = blockchain_selection_
-                .emplace(
-                    std::piecewise_construct,
-                    std::forward_as_tuple(type),
-                    std::forward_as_tuple(
-                        opentxs::factory::BlockchainSelectionModel(
-                            api_, chains, cb)))
-                .first;
+                  .emplace(
+                      std::piecewise_construct,
+                      std::forward_as_tuple(type),
+                      std::forward_as_tuple(
+                          opentxs::factory::BlockchainSelectionModel(
+                              api_, chains, cb)))
+                  .first;
```

Comment:
Instead of that you can try to use try_emplace, it should be more readable then :P

Reply:
clang-format cannot complain if you want to use specific function :P 
No, I would say:
auto it = blockchain_selection_.try_emplace(type, api_, chains, cb).first;

Reply:
It doesn't work because BlockchainSelectionModel returns unique_ptr. My proposition was compiled.